frontend/src/components/DownloadAnalysisLink/DownloadAnalysisLink.tsx

```diff
@@ -44,10 +44,13 @@ function onClick(store: Store) {
     label: selectedDataset ? selectedDataset.name : 'Unknown Dataset'
   })
 
-  if (useStaticCsvs){ // If in production, use the cached static CSVs.
+  if (useStaticCsvs) { // If in production, use the cached static CSVs.
     const staticCsvUrl = getStaticCsvUrl(selectedDataset, method)
     window.open(staticCsvUrl)
   } else { // Otherwise, generate the CSV.
+    if (store.get('serviceAreas').length > 100 && !confirm('Preparing the file for this state may take a couple of minutes. \n\nPress OK to continue.')) {
+      return
```

Comment:
so if they cancel, we dont run the analysis, correct?

Reply:
yep!